plumbing/protocol/packp/filter.go

```diff
@@ -0,0 +1,45 @@
+package packp
+
+import (
+	"fmt"
+	"github.com/go-git/go-git/v5/plumbing"
+	"net/url"
+	"strings"
+)
+
+// Filter values enable the partial clone capability which causes
+// the server to omit objects that match the filter.
+type Filter string
+
+// FilterBlobNone omits all blobs.
+func FilterBlobNone() Filter {
+	return "blob:none"
+}
+
+// FilterBlobLimit omits blobs of size at least n bytes. n can be zero,
+// in which case all blobs will be omitted.
+func FilterBlobLimit(n uint64) Filter {
+	return Filter(fmt.Sprintf("blob:limit=%d", n))
+}
+
+// FilterTreeDepth omits all blobs and trees whose depth from the root tree
+// is larger or equal to depth.
+func FilterTreeDepth(depth uint64) Filter {
+	return Filter(fmt.Sprintf("tree:%d", depth))
+}
+
+// FilterObjectType omits all objects which are not of the requested type t.
+func FilterObjectType(t plumbing.ObjectType) Filter {
```

Comment:
We need to add validation here and return `error` if an unsupported ObjectType (InvalidObject, OFSDeltaObject, REFDeltaObject or AnyObject) is used.

Reply:
Implemented in https://github.com/go-git/go-git/pull/1000/commits/83f87a265c4dd01dbeedfaf99517fa000b4dbd98.